core/src/main/java/io/grpc/internal/HedgingPolicy.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableSet;
+import io.grpc.Status.Code;
+import java.util.Collections;
+import java.util.Set;
+import javax.annotation.ParametersAreNonnullByDefault;
```

Comment:
FYI I don't think this annotation is used anywhere else in the repository. Not sure how useful it will be if its only used in isolation in this one class?

Reply:
I don't have strong opinion on it. Just try to stop using `@Nonnull` annotations. Removed `@ParametersAreNonnullByDefault`